src/lang/test_expected/general_ad.expected

```diff
@@ -0,0 +1,8 @@
+def grad_mm(double(M, K) A, double(K, N) B, double(M, N) O, double(M, N) W, double(M, N) seed_d_O, double(M, N) seed_d_W) -> (d_A, d_B) {
+  d_O(i, j) = seed_d_d_O(i, j)
+  d_O(i, j) += (seed_d_W(i, j) * 2)
+  d_A(i, k) = 0
```

Comment:
In order to be parsed by TC this will need either a where clause or to be folded into `+=!`.

More generally I'd make the test run parsing + sema on the produced TC and print / compare against the canonical TC. Dogfooding would have caught the where case here.